File: Arcade/Intro/Java/AdjacentElementsProduct.java
import java.util.stream.IntStream;

final class AdjacentElementsProduct {
	
	int adjacentElementsProduct(int[] inputArray) {
	    return IntStream.range(1, inputArray.length)
        .map(n -> inputArray[n] * inputArray[n - 1])
        .max()
        .getAsInt();
	}
}

File: Arcade/Intro/javascript/adjacentElementsProduct.js
function adjacentElementsProduct(inputArray) {
    let max = inputArray[0] * inputArray[1];
    
    for(let i = 0; i < inputArray.length - 1; i++) {
        let p = inputArray[i] * inputArray[i + 1];
        if(p > max) max = p;
    }
    
    return max;
}